src/UserInterfaces/WindowsForms/DecompilerUiService.cs

```diff
@@ -84,7 +85,10 @@ public virtual string ShowOpenFileDialog(string fileName)
         public virtual string ShowSaveFileDialog(string fileName)
         {
             if (!string.IsNullOrEmpty(fileName))
-                sfd.FileName = fileName;
+            {
+                sfd.FileName = Path.GetFileName(fileName);
+                sfd.InitialDirectory = Path.GetDirectoryName(fileName);
```

Comment:
Comment: thanks for fixing this. It's been bugging me for ages, and I've never gotten around to taking care of it.